ports/atmel-samd/common-hal/pulseio/PWMOut.c

```diff
@@ -272,35 +285,69 @@ void common_hal_pulseio_pwmout_deinit(pulseio_pwmout_obj_t* self) {
 
 extern void common_hal_pulseio_pwmout_set_duty_cycle(pulseio_pwmout_obj_t* self, uint16_t duty) {
     const pin_timer_t* t = self->timer;
-    uint8_t index;
     if (t->is_tc) {
-        index = timer_index((uint32_t) self->timer->tc);
-        uint16_t adjusted_duty = timer_periods[index] * duty / 0xffff;
-        tc_set_compare_value(&self->tc_instance, t->channel, adjusted_duty);
+        uint16_t adjusted_duty = tc_periods[t->index] * duty / 0xffff;
+        #ifdef SAMD21
+        tc_insts[t->index]->COUNT16.CC[t->wave_output].reg = adjusted_duty;
+        #endif
+        #ifdef SAMD51
+        Tc* tc = tc_insts[t->index];
+        while (tc->COUNT16.SYNCBUSY.bit.CC1 != 0) {
+            // Wait for a previous value to be written. This can wait up to one period so we do
```

Comment:
If the period is really short will we spend too long in `MICROPY_VM_HOOK_LOOP`?

Reply:
We could potentially spend a long time but we don't make any promises about how quickly we'll change duty_cycle so we should be ok.